### PR TITLE
Add image extraction to PDF. Polish code

### PR DESCRIPTION
**Pull request by dSupertramp**
Hi everyone! Danilo's here

I added the extraction of the images from PDF (images are saved in a local folder)

I also cleaned the code a little bit

Thanks for all!

### CODE REVIEW

**Comment by StanGirard** [comment]
Hey @dSupertramp this is awesome! Could you try to use something else than FITZ ? Its license is AGPL, meaning it can't be used in production if you don't have a commercial license with them.

**Comment by dSupertramp** [comment]
> Hey @dSupertramp this is awesome! Could you try to use something else than FITZ ? Its license is AGPL, meaning it can't be used in production if you don't have a commercial license with them.

Done! I used pypdf

**Comment by StanGirard** [comment]
@chloedia @AmineDiro can you test ? :)

**Comment by StanGirard** [comment]
@dSupertramp it seems you forgot to add it to the dependencies ;)

File: Pipfile [hunk 1]
```diff
@@ -19,6 +19,7 @@ poppler-utils = "*"
 langchain-openai = "*"
 langchain-core = "*"
 python-dotenv = "*"
+pypdf = "*"
```
Comment (AmineDiro) [inline Pipfile:22]: Should probably fix the version

File: megaparse/Converter.py [hunk 2]
```diff
@@ -32,45 +34,46 @@ class Converter:
...
             f.write(md_content)
 
+
 class XLSXConverter(Converter):
     def __init__(self) -> None:
         pass
 
-    def convert(self, file_path: str) -> str:
-        xls = pd.ExcelFile(file_path) #type: ignore
+    def convert(self, file_path: Path | str) -> str:
```
Comment (AmineDiro) [inline megaparse/Converter.py:49]: Small suggestion here : should check that path is valid if passed a str
```suggestion
    def convert(self, file_path: Path | str) -> str:
      if isinstance(file_path, str):
           file_path = Path(file_path) 
```

File: megaparse/Converter.py [hunk 3]
```diff
@@ -281,29 +310,29 @@ def save_md(self, md_content: str, file_path: Path | str) -> None:
...
+    @staticmethod
+    def save_images(file_path: Path | str, images_dir: str) -> None:
+        if not os.path.exists(images_dir):
+            os.makedirs(images_dir)
+        reader = PdfReader(file_path)
+        for page_num, page in enumerate(reader.pages, start=1):
+            count = 0
+            for image_file_object in page.images:
+                with open(
+                    os.path.join(images_dir, f"image_{count+1}_page_{page_num}.png"),
```
[the remark this answers is not in the document]
Comment (AmineDiro) [reply megaparse/Converter.py:322]: No need for the count here we could `enumerate(page.images,1)`

**Review by AmineDiro — COMMENTED** [review]
Thanks for the  PR !  Nice work, small remarks on some minor changes.

**Comment by dSupertramp** [comment]
Everything solved!

**Pull request closed without merging**